File: LEVEL1/다트게임.py
def solution(dartResult):
    answer = []
    k = []
    
    for i in range(1, len(dartResult)):
        if dartResult[i - 1].isnumeric():
            n = int(dartResult[i - 1])
            if i < 3:
                if n == 0 and i == 2:
                    n += 10
            else:
                if n == 0 and dartResult[i - 2].isnumeric():
                    n += 10
        
        if dartResult[i] == 'S':
            n = n ** 1
            k.append(n)
        elif dartResult[i] == 'D':
            n = n ** 2
            k.append(n)
        elif dartResult[i] == 'T':
            n = n ** 3
            k.append(n)

    dartResult += 'a'
    
    for i in range(1, len(dartResult) - 1):
        if dartResult[i] == 'S' or dartResult[i] == 'D' or dartResult[i] == 'T':
            tmp = k[len(answer)]
            if dartResult[i + 1] == '#':
                tmp *= (-1)
            elif dartResult[i + 1] == '*':
                tmp *= 2
                if len(answer) > 0:
                    answer.append((answer.pop() * 2))
            answer.append(tmp)            
    
    return sum(answer)

File: KAKAO/LEVEL1/다트게임.js
function solution(dartResult) {
  let answer = [];
  let tmp = 0;

  for (let i = 0; i < dartResult.length; i++) {
    if (dartResult[i] >= 0 && dartResult[i] < 10) {
      if (dartResult[i] === 1 && dartResult[i + 1] === 0) {
        tmp = 10;
        i++;
      } else {
        tmp = dartResult[i] * 1;
      }
    } else if (dartResult[i] === 'S') {
      answer.push(tmp);
    } else if (dartResult[i] === 'D') {
      answer.push(tmp ** 2);
    } else if (dartResult[i] === 'T') {
      answer.push(tmp ** 3);
    } else if (dartResult[i] === '*') {
      answer[answer.length - 1] *= 2;
      answer[answer.length - 2] *= 2;
    } else if (dartResult[i] === '#') {
      answer[answer.length - 1] *= -1;
    }
  }
  return answer.reduce((a, c) => a + c, 0);
}
